packages/gatsby/cache-dir/fast-refresh-overlay/components/build-error.js

```diff
@@ -1,44 +1,35 @@
-import React from "react"
-import Overlay from "./overlay"
-import Anser from "anser"
-import CodeFrame from "./code-frame"
-import { prettifyStack } from "../utils"
+import * as React from "react"
+import { Overlay, Header, Body, Footer, HeaderOpenClose } from "./overlay"
+import { CodeFrame } from "./code-frame"
+import { prettifyStack, openInEditor } from "../utils"
 
-const BuildError = ({ error, open, dismiss }) => {
-  const [file, cause, _emptyLine, ...rest] = error.split(`\n`)
-  const [_fullPath, _detailedError] = rest
-  const detailedError = Anser.ansiToJson(_detailedError, {
-    remove_empty: true,
-    json: true,
-  })
-  const lineNumberRegex = /^[0-9]*:[0-9]*$/g
-  const lineNumberFiltered = detailedError.filter(
-    d => d.content !== ` ` && d.content.match(lineNumberRegex)
-  )[0]?.content
-  const lineNumber = lineNumberFiltered.substr(
-    0,
-    lineNumberFiltered.indexOf(`:`)
-  )
+// Error that is thrown on e.g. webpack errors and thus can't be dismissed and must be fixed
+export function BuildError({ error }) {
+  // Incoming build error shape is like this:
+  // ./relative-path-to-file
+  // Additional information (sometimes empty line => handled in "prettifyStack" function)
+  // /absolute-path-to-file
+  // Errors/Warnings
+  const [file, ...rest] = error.split(`\n`)
   const decoded = prettifyStack(rest)
 
-  const header = (
-    <>
-      <div data-gatsby-overlay="header__cause-file">
-        <p>{cause}</p>
-        <span>{file}</span>
-      </div>
-      <button
-        onClick={() => open(file, lineNumber)}
-        data-gatsby-overlay="header__open-in-editor"
-      >
-        Open in editor
-      </button>
-    </>
+  return (
+    <Overlay>
+      <Header data-gatsby-error-type="build-error">
+        <div data-gatsby-overlay="header__cause-file">
+          <h1 id="gatsby-overlay-labelledby">Failed to compile</h1>
+          <span>{file}</span>
+        </div>
+        <HeaderOpenClose open={() => openInEditor(file, 1)} dismiss={false} />
```

Comment:
This is a follow up: we should put this openInEditor on the same place as runtime errors so people know where to look.